packages/tgpu-cli/gen.mjs

```diff
@@ -1,64 +1,107 @@
 // @ts-check
 
 import fs from 'node:fs/promises';
+import path from 'node:path';
 import { ArrayInfo, StructInfo, TemplateInfo, WgslReflect } from 'wgsl_reflect';
 
 const cwd = new URL(`file:${process.cwd()}/`);
 
 const LENGTH_VAR = 'arrayLength';
 
+export class Context {
+  /**
+   * @param { {toTs: boolean, toCommonJs: boolean} } options
+   */
+  constructor(options) {
+    this.toTs = options.toTs;
+    this.toCommonJs = options.toCommonJs;
+
+    /** @type { string[] } */
+    this.exportsList = [];
+  }
+}
+
 /**
  * @param { string } input
  * @param { string } output
+ * @param { boolean } toTs
+ * @param { boolean } toCommonJs
+ * @param { boolean } overwriteExisting
  */
-async function main(input, output) {
+async function main(input, output, toTs, toCommonJs, overwriteExisting) {
```

Comment:
Tiny nitpick, not fond of this many primitive parameters being passed in as positional parameters. Maybe we could merge the Context and Options into one, and pass that around instead? (this would also get rid of the class in favor of a plain object).

```ts
/**
 * @typedef {object} Options
 * @prop {string} input
 * @prop {string} output
 * @prop {boolean} toTs
 * @prop {'commonjs'|'esmodule'} moduleSyntax
 * @prop {'keep'|'overwrite'} [existingFileStrategy]
 */
```

```suggestion
/**
 * @param {Options} options
 */
async function main(options) {
```